heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java

```diff
@@ -315,18 +315,17 @@ public static void main(String[] args) throws Exception {
                 topologyPackage, topologyBinaryFile, topologyDefnFile, topology))
             .build());
 
-    LOG.fine("Static config loaded successfully ");
+    LOG.fine("Static config loaded successfully");
     LOG.fine(config.toString());
 
     SubmitterMain submitterMain = new SubmitterMain(config, topology);
-    boolean isSuccessful = submitterMain.submitTopology();
-
-    // Log the result and exit
-    if (!isSuccessful) {
+    try {
+      submitterMain.submitTopology();
+    } catch (TopologySubmissionException e) {
+      System.out.println(e.getMessage());
```

Comment:
what do you mean by "this logic"? And I agree we can have a super-class of those exceptions.